### DISCUSSION TITLE
Attribute error in importing the Boost external motor

### DISCUSSION THREAD

**Discussion by vjchan99** [pybricks/support#1113] [General]
Hi, I keep getting an attribute error which I think is related to the following part of the code:

```python
from pybricks.hubs import MoveHub
from pybricks.pupdevices import Motor, Remote
from pybricks.parameters import Button, Color, Direction, Port, Side, Stop
from pybricks.robotics import DriveBase
from pybricks.tools import wait, StopWatch
```

I am using the external motor from the Lego Boost set. Is there a problem with the code?

Thanks.

**Comment by dlech** [comment]
Which firmware version are you using?

What is the exact error you are getting?

**Reply by vjchan99** [reply]
I'm getting the following error:
File "bot.py", in <module>
AttributeError: no such attribute

**Reply by dlech** [reply]
Is what you shared the full program? 

If there was a problem with the imports, I think it would be an ImportError rather than an AttributeError.

**Reply by vjchan99** [reply]
No it isn't. Here's the whole program:
----------------------------------------------------------------------------
```python
from pybricks.hubs import MoveHub
from pybricks.pupdevices import Motor, Remote
from pybricks.parameters import Button, Color, Port

# Define objects for a simple two motor rover
hub = MoveHub()
left_motor = Motor(Port.A)
right_motor = Motor(Port.B)

# Connect to the remote
hub.light.on(Color.YELLOW)    # Turn hub light yellow while trying to connect
rc = Remote()                 # Connect to the remote
hub.light.on(Color.GREEN)     # Set hub light to green to indicate connection

# Speeds to use
STRAIGHT_SPEED = 2000    # Straight driving speed (motor power)
TURN_POWER = 100       # Turning power (motor power)

# The main loop repeatedly tests the remote buttons and reacts
while True:
    # Get the set of buttons that are currently pressed
    pressed = rc.buttons.pressed()

    # Determine what straight driving power to use
    if Button.LEFT_PLUS in pressed:
        left_motor.dc(STRAIGHT_SPEED)
        right_motor.dc(STRAIGHT_SPEED)
    elif Button.LEFT_MINUS in pressed:
        left_motor.dc(-STRAIGHT_SPEED)
        right_motor.dc(-STRAIGHT_SPEED)
    else:
        left_motor.stop()
        right_motor.stop()

    # Determine what turn power to use
    if Button.RIGHT_PLUS in pressed:
        left_motor.dc(-TURN_POWER)
        right_motor.dc(TURN_POWER)
    elif Button.RIGHT_MINUS in pressed:
        left_motor.dc(TURN_POWER)
        right_motor.dc(-TURN_POWER)
    else:
        left_motor.stop()
        right_motor.stop()
```
----------------------------------------------------------------------------

The above program works just fine. But I get the error when I add the following functionality:
-import motor to port D
- Each time either of the red buttons on the remote are pressed, make the motor move 150 degrees at a speed of 100 and immediately move -160 degrees at a speed of 100.

**Reply by dlech** [reply]
> -import motor to port D

Is this literally the line of code that you are adding? It should be

```python
other_motor = Motor(Port.D)
```

**Reply by dlech** [reply]
> * Each time either of the red buttons on the remote are pressed,

The program only uses the "+" and "-" buttons so I don't see how pressing the red button could do anything.

**Reply by vjchan99** [reply]
Sorry I didn't realise that. I had written the following, which is probably what is triggering the error "AttributeError: no such attribute":

    # Check if either of the red buttons is pressed
    if Button.LEFT_RED in pressed or Button.RIGHT_RED in pressed:
        extra_motor.run_angle(EXTRA_SPEED, DEGREE)
        extra_motor.run_angle(-EXTRA_SPEED, -DEGREE)

I've used the left + & - to move forward and back and the right ones to turn. I was hoping I could assign something else on the remote to control the external motor on Port D. Any suggestions?

**Reply by dlech** [reply]
It looks like our docs could use some diagrams showing which button corresponds to which value. https://docs.pybricks.com/en/latest/parameters/button.html

There isn't a `Button.LEFT_RED`. It is just `Button.LEFT`.

**Reply by vjchan99** [reply]
Thanks. To clarify, the red buttons can be used as controls but they are expressed as Button.LEFT and Button.RIGHT correct?

**Reply by dlech** [reply]
Correct.

**Reply by vjchan99** [reply]
Thanks so much for your help!

**Reply by dlech** [reply]
You are welcome. :smile:
